client/src/entrypoints/admin/privacy-switch.js

```diff
@@ -1,32 +1,49 @@
-/* global ModalWorkflow */
+import { domReady } from '../../utils/domReady';
+import { encodeForm } from '../../utils/encodeForm';
 
-import $ from 'jquery';
+/**
+ * Initializes the privacy switch functionality.
+ * Attaches event listeners to privacy trigger buttons to open the ModalWorkflow.
+ */
+function initPrivacySwitch() {
+  const privacyTriggers = document.querySelectorAll(
+    '[data-a11y-dialog-show="set-privacy"]',
+  );
 
-$(() => {
-  /* Interface to set permissions from the explorer / editor */
-  $('[data-a11y-dialog-show="set-privacy"]').on('click', function setPrivacy() {
-    ModalWorkflow({
-      dialogId: 'set-privacy',
-      url: this.getAttribute('data-url'),
-      onload: {
-        set_privacy(modal) {
-          $('form', modal.body).on('submit', function handleSubmit() {
-            modal.postForm(this.action, $(this).serialize());
-            return false;
-          });
-        },
-        set_privacy_done(modal, { is_public: isPublic }) {
-          document.dispatchEvent(
-            new CustomEvent('w-privacy:changed', {
-              bubbles: true,
-              cancelable: false,
-              detail: { isPublic },
-            }),
-          );
-          modal.close();
+  privacyTriggers.forEach((trigger) => {
+    trigger.addEventListener('click', (event) => {
+      event.preventDefault();
```

Comment:
It might be worth finding call usage of the data attribute in the html. 

As long as they all have `type="button" ` we wont need to prevent default.